src/python/pants/base/exception_sink.py

```diff
@@ -252,6 +247,32 @@ def _recapture_fatal_error_log_streams(cls, new_log_location):
 
     return (pid_specific_error_stream, shared_error_stream)
 
+  @classmethod
+  def reset_graceful_signal_handler(cls, graceful_signal_handler):
```

Comment:
Naming quibble. You do not care that `graceful_signal_handler` is graceful and it won't be in any case where it's subclassed. You only care that you have a `GracefulSignalHandler` instance for its handlers, not their gracefullness. Consider `s/GracefulSignalHandler/SignalHandler/` and `s/graceful_signal_handler/signal_handler/g` and just leave the graceful bit as a doc comment in `SignalHandler` that the default behavior is graceful for handler methods not over-ridden.

Reply:
I performed the text substitutions you described and briefly described the graceful bit in the docstring!